### PR TITLE
[ML examples] Introduce ModelRunner

### PR DESCRIPTION
**Pull request by artemrakhov**
Sometimes it's useful to load a model and run it without any additional user inputs. Simply use whatever weights/inputs are written in init_net.

### CODE REVIEW

**Comment by nadavrot** [comment]
Can we add this as a flag to the existing loader?

**Comment by artemrakhov** [comment]
@nadavrot: we can. But I thought it's a bit weird, that `image-classifier` binary would have a secret mode, where it doesn't load any images, doesn't classify anything and has different code path for printing output. When we support loading of translation models, will we also run them through this binary? I assumed it's better to have separate binary for every type of models.

I'll do this: implement this load-path in the existing binary in case if number of positional args is 0, rename the binary to `model-runner`, and create a PR. Probably the added complexity won't be too bad.

**Comment by nadavrot** [comment]
Ok. I don’t have a strong opinion here. I you think that adding a new binary is better then go for it.

File: tools/loader/ModelRunner.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                             *loader.getFunction());
+  }
+  output = LD->getRoot();
+
+  // Compile the model, and perform quantization/emit a bundle/dump debug info
+  // if requested from command line.
+  loader.compile();
+
+  // If in bundle mode, do not run inference.
+  if (!emittingBundle()) {
```
Comment (qcolombet) [inline tools/loader/ModelRunner.cpp]: Out-of-curiosity, what are the externally observable effect if emit-bundle is not set?
Basically, I am wondering if we want this binary to do something externally visible to tell the user that it did something. Like printing loading and compilation succeeded.

File: tools/loader/ModelRunner.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+using namespace glow;
+
+int main(int argc, char **argv) {
+  // The loader verifies/initializes command line parameters, and initializes
+  // the ExecutionEngine and Function.
+  Loader loader(argc, argv);
+
+  // Create the model based on the input net, and get SaveNode for the output.
+  SaveNode *output;
```
Comment (rdzhabarov) [inline tools/loader/ModelRunner.cpp]: super nit: better to merge declaration and init below.

File: tools/loader/ModelRunner.cpp (hunk 1 above)
Comment (artemrakhov) [reply tools/loader/ModelRunner.cpp]: If emit-bundle is not set, then the branch gets executed and obviously something gets printed.

If emit-bundle is set, then unfortunately there's no output printed to stdout. We may want to fix it in follow up PR. Note, that the fix will be in `Loader.cpp` file, which is shared between ModelRunner and ImageClassifier.

File: tools/loader/ModelRunner.cpp (hunk 2 above)
Comment (artemrakhov) [reply tools/loader/ModelRunner.cpp]: Oops, fixed.

**Comment by artemrakhov** [comment]
I've decided to go with separate binaries. One reason is that we'll want different flags for different modes. E.g. one would need to manually verify that if we're in text mode then image flags are not set, etc. In future, we may also have video flags and audio flags.

Loader class already does good job by hiding common code. I'll see if I can refactor it a bit more to have one-liner for whole `initModelLoaderAndCompile` thing.

**Pull request merged**